### PR TITLE
Add action button to dismiss variants from list #2004

### PR DESCRIPTION
**Pull request by elevu**
This PR adds the column "Dismiss variant" to all the variants tables.

Aug-17-2020 10-19-35

**How to test**:
1. Deploy this branch to scout-stating, look at the variant tables for a "Rare disease variant" and then for a "Cancer" variant. Try out the filtering of the table as well since I changed the form a little bit.

**Expected outcome**:
The functionality should be working

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by northwestwitch** [comment]
Nice! I can review as soon as the test pass!

File: scout/server/blueprints/variant/templates/variant/buttons.html [hunk 1]
```diff
@@ -61,6 +61,25 @@
   </form>
 {% endmacro %}
 
+{% macro dismiss_variant_table_button(variant, institute, case, dismiss_variant_options) %}
```
Comment (northwestwitch) [inline scout/server/blueprints/variant/templates/variant/buttons.html:64]: 👍

File: scout/server/blueprints/variants/views.py [hunk 2]
```diff
@@ -153,12 +159,15 @@ def variants(institute_id, case_name):
     if request.form.get("export"):
         return controllers.download_variants(store, case_obj, variants_query)
 
+    dismiss_options = return_dismissed_variants(case_obj.get("track"))
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/views.py:162]: You don't need to invoke a function here, you could just provide the DISMISS_VARIANT_OPTIONS when you return the dictionary (line 170)

File: scout/server/blueprints/variants/views.py [hunk 3]
```diff
@@ -192,10 +201,13 @@ def str_variants(institute_id, case_name):
             ("position", pymongo.ASCENDING),
         ]
     )
+    dismiss_options = return_dismissed_variants(case_obj.get("track"))
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/views.py:204]: Same here

File: scout/server/blueprints/variants/views.py [hunk 4]
```diff
@@ -153,12 +159,15 @@ def variants(institute_id, case_name):
...
 
+    dismiss_options = return_dismissed_variants(case_obj.get("track"))
+
     data = controllers.variants(store, institute_obj, case_obj, variants_query, page)
     return dict(
         institute=institute_obj,
         case=case_obj,
         form=form,
         manual_rank_options=MANUAL_RANK_OPTIONS,
+        dismiss_variant_options=dismiss_options,
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/views.py:170]:
```suggestion
        dismiss_variant_options=DISMISS_VARIANT_OPTIONS,
```

File: scout/server/blueprints/variants/views.py (hunk 2 above)
Comment (northwestwitch) [inline scout/server/blueprints/variants/views.py:162]:
```suggestion
```

File: scout/server/blueprints/variants/views.py (hunk 3 above)
Comment (northwestwitch) [inline scout/server/blueprints/variants/views.py:204]:
```suggestion
```

File: scout/server/blueprints/variants/views.py [hunk 5]
```diff
@@ -192,10 +201,13 @@ def str_variants(institute_id, case_name):
             ("position", pymongo.ASCENDING),
         ]
     )
+    dismiss_options = return_dismissed_variants(case_obj.get("track"))
+
     data = controllers.str_variants(store, institute_obj, case_obj, variants_query, page)
     return dict(
         institute=institute_obj,
         case=case_obj,
+        dismiss_variant_options=dismiss_options,
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/views.py:210]:
```suggestion
        dismiss_variant_options= DISMISS_VARIANT_OPTIONS,
```

File: scout/server/blueprints/variants/views.py [hunk 6]
```diff
@@ -235,9 +247,12 @@ def sv_variants(institute_id, case_name):
 
     data = controllers.sv_variants(store, institute_obj, case_obj, variants_query, page)
 
+    dismiss_options = return_dismissed_variants(case_obj.get("track"))
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/views.py:250]:
```suggestion
```

File: scout/server/blueprints/variants/views.py [hunk 7]
```diff
@@ -235,9 +247,12 @@ def sv_variants(institute_id, case_name):
...
+
     return dict(
         institute=institute_obj,
         case=case_obj,
+        dismiss_variant_options=dismiss_options,
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/views.py:255]:
```suggestion
        dismiss_variant_options= DISMISS_VARIANT_OPTIONS,
```

File: scout/server/blueprints/variants/views.py [hunk 8]
```diff
@@ -310,9 +325,12 @@ def cancer_variants(institute_id, case_name):
         store, institute_id, case_name, variants_query, form, page=page
     )
 
+    dismiss_options = return_dismissed_variants(case_obj.get("track"))
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/views.py:328]:
```suggestion
```

File: scout/server/blueprints/variants/views.py [hunk 9]
```diff
@@ -310,9 +325,12 @@ def cancer_variants(institute_id, case_name):
...
+
     return dict(
         variant_type=variant_type,
         cytobands=cytobands,
+        dismiss_variant_options=dismiss_options,
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/views.py:333]:
```suggestion
        dismiss_variant_options={**DISMISS_VARIANT_OPTIONS, **CANCER_SPECIFIC_VARIANT_DISMISS_OPTIONS},
```

File: scout/server/blueprints/variants/views.py [hunk 10]
```diff
@@ -351,9 +369,12 @@ def cancer_sv_variants(institute_id, case_name):
 
     data = controllers.sv_variants(store, institute_obj, case_obj, variants_query, page)
 
+    dismiss_options = return_dismissed_variants(case_obj.get("track"))
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/views.py:372]:
```suggestion
```

File: scout/server/blueprints/variants/views.py [hunk 11]
```diff
@@ -351,9 +369,12 @@ def cancer_sv_variants(institute_id, case_name):
...
+
     return dict(
         institute=institute_obj,
         case=case_obj,
+        dismiss_variant_options=dismiss_options,
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/views.py:377]:
```suggestion
        dismiss_variant_options={**DISMISS_VARIANT_OPTIONS, **CANCER_SPECIFIC_VARIANT_DISMISS_OPTIONS},
```

File: scout/server/blueprints/variants/templates/variants/cancer-sv-variants.html [hunk 12]
```diff
@@ -37,28 +39,30 @@
...
-      </div>
-      <!--Expand filters form if filters were used in a previous POST request-->
-      <div id="collapseFilters" class="card-body panel-collapse collapse {{ 'show' if expand_search == 'True' }}">
-        <div class="card-body">{{ sv_filters(form, institute, case) }}</div>
-      </div>
-      <div class="card-footer text-center py-3">
-        {% if "all_variants" in session %}
-          Filtered {{session['filtered_variants']}} / {{session["all_variants"]}} variants
-        {% endif %}
+    <form method="POST" id="filters_form" action="{{url_for('variants.cancer_sv_variants', institute_id=institute._id, case_name=case.display_name)}}" >
```
[the remark this answers is not in the document]
Comment (elevu) [reply scout/server/blueprints/variants/templates/variants/cancer-sv-variants.html:43]: I moved the filter form tag inside the outer page div. I had to do it to be able to add another form for the variant dismiss action.

File: scout/server/blueprints/variants/templates/variants/sv-variants.html [hunk 13]
```diff
@@ -38,39 +40,40 @@
...
-      <!--Expand filters form if filters were used in a previous POST request-->
-      <div id="collapseFilters" class="card-body panel-collapse collapse {{ 'show' if expand_search == 'True' }}">
-        <div class="card-body">{{ sv_filters(form, institute, case) }}</div>
-      </div>
-      <div class="card-footer text-center py-3">
-        {% if "all_variants" in session %}
-          Showing {{session['filtered_variants']}} / {{session["all_variants"]}} variants
-        {% endif %}
-      </div>
+    <form method="POST" id="filters_form" action="{{url_for('variants.sv_variants', institute_id=institute._id, case_name=case.display_name)}}"
```
[the remark this answers is not in the document]
Comment (elevu) [reply scout/server/blueprints/variants/templates/variants/sv-variants.html:44]: Same as the other file, I moved the filter form tag inside the outer page div.

File: scout/server/blueprints/variants/views.py [hunk 14]
```diff
@@ -435,3 +456,9 @@ def download_verified():
         )
     flash("No verified variants could be exported for user's institutes", "warning")
     return redirect(request.referrer)
+
+
+def return_dismissed_variants(case_track, dismiss_options=DISMISS_VARIANT_OPTIONS):
```
[the remark this answers is not in the document]
Comment (elevu) [reply scout/server/blueprints/variants/views.py:461]: great! Will do that

File: scout/server/blueprints/variants/templates/variants/sv-variants.html (hunk 13 above)
[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/blueprints/variants/templates/variants/sv-variants.html:44]: Ah ok!

**Pull request merged**